content/blogposts/2024/gp-exploring-data-visualization-with-d3-cody.md

```diff
@@ -0,0 +1,326 @@
+
+On a final bit, we append our SVG node to the DOM so it reflects on our browser.  
+
+```javascript
+let chart = document.getElementById("chart");
+
+chart.append(svg.node());
+
+<div id="chart"></div> //html
+```
+
+Here’s the full code: 
+
+```javascript
+const width = 600; 
+const height = 200;
+
+const values = [100, 90, 80, 60, 50, 40, 30, 20, 10];
+const svg = d3.create("svg")
+    .attr("width", width)
+    .attr("height", height);
+
+svg.selectAll("rect")
+    .data(values)
+    .enter()
+    .append("rect")
+    .attr("x", (d, i) => i * 60)
+    .attr("y", d => height - d)
+    .attr("width", 50)
+    .attr("height", d => d)
+    .attr("fill", "purple");
+
+let chart = document.getElementById("chart");
+
+chart.append(svg.node());
+
+<div id="chart"></div>
+```
+
+Here’s the result:
+
+<Figure
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/gp-exploring-data-visualization-with-d3-cody/image_005.png"
+/>
+
+
+Here, we see the chart is in descending order, just as we arranged the numbers in the array. How about toggling the codebase so our chart appears in ascending order? We can always call on our good old friend, Cody, to help. Let’s give it a try:
+
+<Figure
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/gp-exploring-data-visualization-with-d3-cody/image_006.gif"
+  caption="Using Cody to arrange our chart in ascending order."
+/>
+
+And there! I simply tell Cody to _modify this code so the chart appears in ascending order._ It takes in the highlighted piece of code as context and then provides a context-aware solution by rearranging the values of the numbers of the array. Let’s see how the result looks:
+
+<Figure
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/gp-exploring-data-visualization-with-d3-cody/image_007.png"
+/>
+
+It acts just as we expect it to, in ascending order. I have to say a big thank you to Cody for all the help. 
+
+## Animation and transition in D3.js
+
+When talking about appealing and eye-catching visualization, we must also talk about animation. Animations, when used correctly, add that extra bit of spice to our visualization and make it better. In this section, we will talk about how to apply animation to your visualization in D3.js. 
+
+### The `.transition()` method
+
+This handles the basic functionality of making elements move in D3.js. Without transitions, all elements appear instantaneously on your web browser when loaded or reloaded. Transitions give the flexibility to control the progression of elements on the browser. With the `.transition` method in D3.js, we can monitor the current state of elements and the target state. So, rather than only working with a single(final) state, we’re working with two states here. The element then moves in tune with the difference in values of the two states. We’ll see how that works in a bit. 
```

Comment:
```suggestion
This handles the basic functionality of making elements move in D3.js. Without transitions, all elements appear instantaneously on your web browser when loaded or reloaded. Transitions give the flexibility to control the progression of elements on the browser. With the `.transition` method in D3.js, we can monitor the current state of elements and the target state. So, rather than only working with a single (final) state, we’re working with two states here. The element then moves in tune with the difference in values of the two states. We’ll see how that works in a bit. 
```